runtime/vm/jvminit.c

```diff
@@ -971,6 +971,12 @@ freeJavaVM(J9JavaVM * vm)
 			vm->checkpointState.hookRecords = NULL;
 		}
 
+		J9Pool *classIterationRestoreHookRecords = vm->checkpointState.classIterationRestoreHookRecords;
```

Comment:
This is not legal C code: variables must be declared at the beginning of a block.